nestkernel/stimulating_device.cpp

```diff
@@ -0,0 +1,190 @@
+/*
+ *  stimulating_device.cpp
+ *
+ *  This file is part of NEST.
+ *
+ *  Copyright (C) 2004 The NEST Initiative
+ *
+ *  NEST is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  NEST is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+
+// Includes from nestkernel:
+#include "stimulating_device.h"
+#include "kernel_manager.h"
+
+
+nest::StimulatingDevice::StimulatingDevice()
+  : DeviceNode()
+  , Device()
+  , first_syn_id_( invalid_synindex )
+  , backend_params_( new Dictionary )
+{
+}
+
+nest::StimulatingDevice::StimulatingDevice( StimulatingDevice const& sd )
+  : DeviceNode( sd )
+  , Device( sd )
+  , first_syn_id_( invalid_synindex ) // a new instance can have no connections
+  , backend_params_( sd.backend_params_ )
+{
+  P_ = Parameters_( sd.P_ );
```

Comment:
Can this be moved up to the other initialisations, as `P_( sd.P_ )`?

Reply:
Done